dist/_SassyCast.scss

```diff
@@ -1,188 +1,254 @@
+/// Toggle strict mode, in which script will throw when not able to cast a value
+/// into a certain type (mostly color and number). In non-strict mode, it will
+/// return the default value from the given type.
+/// @access public
+/// @type Bool
+$sc-strict-mode: false !default;
+
+/// Default return value for the `to-number(..)` function when running in
+/// non-strict mode.
+/// @access public
+/// @type Number
+$sc-non-strict-default-number: 0 !default;
+
+/// Default return value for the `to-color(..)` function when running in
+/// non-strict mode.
+/// @access public
+/// @type Number
+$sc-non-strict-default-color: transparent !default;
+
+/// Internal map for dynamically accessing default values for non-strict mode.
+/// @access private
+/// @type Map
+$sc-non-strict-defaults: (
+  'number': $sc-non-strict-default-number,
+  'color': $sc-non-strict-default-color,
+);
+
+/// Internal constants map.
+/// @access private
+/// @type Map
+$sc-constants: (
+  'DECIMAL_SPACE': ('0', '1', '2', '3', '4', '5', '6', '7', '8', '9'),
+  'HEXADECIMAL_SPACE': ('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'),
+  'COLOR_KEYWORDS': (transparent, aliceblue, antiquewhite, aqua, aquamarine, azure, beige, bisque, black, blanchedalmond, blue, blueviolet, brown, burlywood, cadetblue, chartreuse, chocolate, coral, cornflowerblue, cornsilk, crimson, cyan, darkblue, darkcyan, darkgoldenrod, darkgray, darkgreen, darkgrey, darkkhaki, darkmagenta, darkolivegreen, darkorange, darkorchid, darkred, darksalmon, darkseagreen, darkslateblue, darkslategray, darkslategrey, darkturquoise, darkviolet, deeppink, deepskyblue, dimgray, dimgrey, dodgerblue, firebrick, floralwhite, forestgreen, fuchsia, gainsboro, ghostwhite, gold, goldenrod, gray, green, greenyellow, grey, honeydew, hotpink, indianred, indigo, ivory, khaki, lavender, lavenderblush, lawngreen, lemonchiffon, lightblue, lightcoral, lightcyan, lightgoldenrodyellow, lightgray, lightgreen, lightgrey, lightpink, lightsalmon, lightseagreen, lightskyblue, lightslategray, lightslategrey, lightsteelblue, lightyellow, lime, limegreen, linen, magenta, maroon, mediumaquamarine, mediumblue, mediumorchid, mediumpurple, mediumseagreen, mediumslateblue, mediumspringgreen, mediumturquoise, mediumvioletred, midnightblue, mintcream, mistyrose, moccasin, navajowhite, navy, oldlace, olive, olivedrab, orange, orangered, orchid, palegoldenrod, palegreen, paleturquoise, palevioletred, papayawhip, peachpuff, peru, pink, plum, powderblue, purple, rebeccapurple, red, rosybrown, royalblue, saddlebrown, salmon, sandybrown, seagreen, seashell, sienna, silver, skyblue, slateblue, slategray, slategrey, snow, springgreen, steelblue, tan, teal, thistle, tomato, turquoise, violet, wheat, white, whitesmoke, yellow, yellowgreen),
```

Comment:
Might be good to do the string conversion here so that it's not done every time `to-color()` is called